File: frontend/zeal-edu/src/app/core/models/exam-result.ts
export interface ExamResult {
  resultId: string;
  examId: string;
  enrollmentId: string;
  candidateCode: string;
  candidateFullName: string;
  score: number;
  grade: string | null;
  isPassed: boolean;
  isFinalized: boolean;
  gradedById: string;
  gradedByName: string;
  isOverridden: boolean;
  overrideById: string | null;
  overrideByName: string | null;
  overrideReason: string | null;
  gradedAt: string;
}

export interface OverrideExamResultPayload {
  score: number;
  overrideReason: string;
}

export interface ExamResultsQuery {
  page: number;
  pageSize: number;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
}

File: backend/src/ZealEducation.Domain/Entities/ExamResult.cs
using ZealEducation.Domain.Common;

namespace ZealEducation.Domain.Entities;

public class ExamResult : BaseAuditableEntity
{
    public Guid ExamId { get; set; }
    public Guid EnrollmentId { get; set; }
    public decimal Score { get; set; }
    public string? Grade { get; set; }
    public bool IsPassed { get; set; }
    public bool IsFinalized { get; set; }
    public Guid GradedById { get; set; }
    public bool IsOverridden { get; set; }
    public Guid? OverrideById { get; set; }
    public string? OverrideReason { get; set; }
    public DateTime GradedAt { get; set; }

    public Examination Examination { get; set; } = default!;
    public Enrollment Enrollment { get; set; } = default!;
    public Staff GradedBy { get; set; } = default!;
    public Staff? OverrideBy { get; set; }
}
